### PR TITLE
docs: add adid to rn and update experiment ios integration

### PR DESCRIPTION
**Pull request by liuyang1520**
# Dev Docs PR
## Description
- docs: add adid to rn and update experiment ios integration

image
image

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Review by dnshi — APPROVED** [review]

**Review by bgiori — APPROVED** [review]
LGTM, thanks for updating this!

File: docs/experiment/sdks/android-sdk.md [hunk 1]
```diff
@@ -159,6 +159,8 @@ If you use either Amplitude or Segment Analytics SDKs to track events into Ampli
         )
         ```
 
+    Note that, if you are using a custom instance name for analytics, you will need to set the same value in the `instanceName` [configuration option](#configuration) in the experiment SDK.
```
Comment (dnshi) [inline docs/experiment/sdks/android-sdk.md:162]: It would be great to mention that the `instanceName` is case insensitive.

Comment (liuyang1520) [reply docs/experiment/sdks/android-sdk.md:162]: Thanks! Updated!

File: docs/experiment/sdks/android-sdk.md [hunk 2]
```diff
@@ -286,7 +286,7 @@ The SDK client can be configured once on initialization.
     | `automaticFetchOnAmplitudeIdentityChange` | Only matters if you use the `initializeWithAmplitudeAnalytics` initialization function to seamlessly integrate with the Amplitude Analytics SDK. If `true` any change to the user ID, device ID or user properties from analytics will trigger the experiment SDK to fetch variants and update it's cache. | `false` |
     | `userProvider` | An interface used to provide the user object to `fetch()` when called. See [Experiment User](https://developers.experiment.amplitude.com/docs/experiment-user#user-providers) for more information. | `null` |
     | `exposureTrackingProvider` | Implement and configure this interface to track exposure events through the experiment SDK, either automatically or explicitly. | `null` |
-    | `instanceName` | Custom instance name for experiment SDK instance. | `null` |
+    | `instanceName` | Custom instance name for experiment SDK instance. The value of this field is case-insensitive. | `null` |
```
Comment (dnshi) [inline docs/experiment/sdks/android-sdk.md:289]: Maybe we can highlight it with bold font?

**Review by dnshi — APPROVED** [review]

**Review by caseyamp — APPROVED** [review]

**Pull request merged**